Migrate ci to git hub actions

gotanygrapes-lemonade opened the pull request:
### Context

We have [a sensible default](https://docs.envato.net/architecture-guild/sensible-defaults/expected/open-source-ci.html) to **not** use our internal Buildkite agents to run CI for open-source projects.

However, we're doing that for a number of open-source Buildkite plugins. (Probably historical, before the default was in-place.) Let's update these projects to use GitHub Actions.


### Considerations

An [example GitHub Actions configuration](https://github.com/envato/heroku-container-deploy-buildkite-plugin/blob/16f73f647bbb6e5cca4217ec5413e6e3d3bb2f41/.github/workflows/test.yml) we can use for inspiration (or copy).

- Adding a YAML file to the `.github/workflows/` directory should be enough to enable GitHub actions
- Delete Buildkite pipelines by navigating to their settings page in a browser and click the `Delete Pipeline` in the pipeline management section

Commit:
create tests.yml to use Github Actions

Commit:
add GitHub Actions badge to README

orien commented on README.md:
```diff
@@ -1,6 +1,6 @@
 # ECS Scheduled Tasks Buildkite Plugin
 
-A [Buildkite plugin](https://buildkite.com/docs/agent/v3/plugins) for deploying [Amazon ECS Scheduled Tasks](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/scheduling_tasks.html).
```
Let's not delete this line.

orien reviewed (approved):
👍  The GitHub Actions build is failing, but I see it was failing on Buildkite as well.

Let's delete the `.buildkite` directory also.

gotanygrapes-lemonade committed:
Update README.md

runlevel5 reviewed (approved):
LGTM

P/S: Please get all failed tests addressed

Commit:
remove .buildkite directory

Commit:
Merge branch 'migrate_CI_to_GitHubActions' of https://github.com/envato/ecs-scheduled-tasks-buildkite-plugin into migrate_CI_to_GitHubActions

viraptor committed:
Fix the test variables to match the plugin

The tests were out of sync with the actual plugin variables. The
`ECS_DEPLOY` (probably copy-paste from another project) should've been
`ECS_SCHEDULED_TASKS`.

Commit:
fix tests: Lint and ShellCheck
- Lint: add the plugin name
- ShellCheck: modified hooks/command, declare and assign separately

Commit:
Merge branch 'migrate_CI_to_GitHubActions' of https://github.com/envato/ecs-scheduled-tasks-buildkite-plugin into migrate_CI_to_GitHubActions

Commit:
declare and assign separately

Commit:
try pass filenames to pass tests (error: SC2261)

Commit:
revert changes to pass filenames

Commit:
attempt to fix shellcheck error: merge two jsons

Commit:
attempt to fix shellcheck error: merge two jsons

Commit:
attempt to fix shellcheck error: merge two jsons

Commit:
attempt to fix shellcheck error: merge two jsons

orien commented on tests/command.bats:
```diff
@@ -2,19 +2,19 @@
 load '/usr/local/lib/bats/load.bash'
 
-# Uncomment to enable stub debug output:
+# Uncgomment to enable stub debug output:
```
This looks unintentional.

Bing-Violet replied:
oops, will fix it. Thanks Orien.

orien reviewed (approved):
Nice one.

gotanygrapes-lemonade committed:
Update command.bats

remove the typo
